pkg/build/iso_test.go

```diff
@@ -65,42 +93,68 @@ func TestDeleteExistingImage(t *testing.T) {
 	require.True(t, os.IsNotExist(err))
 }
 
-func TestCreateXorrisoCommand(t *testing.T) {
+func TestWriteIsoScript_Extract(t *testing.T) {
 	// Setup
-	builder := Builder{
-		context: &image.Context{
-			ImageConfigDir: "config-dir",
-			CombustionDir:  "combustion",
-			ImageDefinition: &image.Definition{
-				Image: image.Image{
-					BaseImage:       "base-image",
-					OutputImageName: "build-image",
-				},
-			},
-		},
-	}
+	ctx, teardown := setupContext(t)
+	defer teardown()
+	builder := Builder{context: ctx}
 
 	// Test
-	cmd, logfile, err := builder.createXorrisoCommand()
+	err := builder.writeIsoScript(extractIsoTemplate, extractIsoScriptName)
 
 	// Verify
 	require.NoError(t, err)
 
-	defer func() {
-		assert.NoError(t, os.Remove(logfile.Name()))
-	}()
+	expectedFilename := filepath.Join(ctx.BuildDir, extractIsoScriptName)
+	_, err = os.Stat(expectedFilename)
+	require.NoError(t, err)
+
+	foundBytes, err := os.ReadFile(expectedFilename)
+	require.NoError(t, err)
+	found := string(foundBytes)
+
+	expectedIsoExtractDir := filepath.Join(ctx.BuildDir, isoExtractDir)
+	assert.Contains(t, found, fmt.Sprintf("ISO_EXTRACT_DIR=%s", expectedIsoExtractDir))
+
+	expectedRawExtractDir := filepath.Join(ctx.BuildDir, rawExtractDir)
+	assert.Contains(t, found, fmt.Sprintf("RAW_EXTRACT_DIR=%s", expectedRawExtractDir))
+
+	expectedIsoPath := builder.generateBaseImageFilename()
+	assert.Contains(t, found, fmt.Sprintf("ISO_SOURCE=%s", expectedIsoPath))
+}
+
+func TestWriteIsoScript_Rebuild(t *testing.T) {
```

Comment:
Before you yell at me and my distaste for table tests, I did consider them here. Ultimately though, we're going to further modify the rebuild script for other functionality and I suspect the testing for those changes will further distinguish the two tests. That said, if those changes aren't as invasive as I fear, I'll refactor to table tests.